source/collection-management/tests/clustered-indexes.yml

```
@@ -0,0 +1,102 @@
+description: "clustered-indexes"
+
+schemaVersion: "1.0"
+
+runOnRequirements:
+  - minServerVersion: "5.3"
+
+createEntities:
+  - client:
+      id: &client0 client0
+  - database:
+      id: &database0 database0
+      client: *client0
+      databaseName: &database0Name ts-tests
+  - collection:
+      id: &collection0 collection0
+      database: *database0
+      collectionName: &collection0Name test
+
+initialData:
+  - collectionName: *collection0Name
+    databaseName: *database0Name
+    documents: []
+
+tests:
+  - description: "createCollection with clusteredIndex"
+    operations:
+      - name: dropCollection
+        object: *database0
+        arguments:
+          collection: *collection0Name
+      - name: createCollection
+        object: *database0
+        arguments:
+          collection: *collection0Name
+          clusteredIndex:
+            key: { _id: 1 }
+            unique: true
+            name: &index0name "test index"
+      - name: assertCollectionExists
+        object: testRunner
+        arguments:
+          databaseName: *database0Name
+          collectionName: *collection0Name
+
+  - description: "listCollections includes clusteredIndex"
+    operations:
+      - name: dropCollection
+        object: *database0
+        arguments:
+          collection: *collection0Name
+      - name: createCollection
+        object: *database0
+        arguments:
+          collection: *collection0Name
+          clusteredIndex:
+            key: { _id: 1 }
+            unique: true
+            name: &index0name "test index"
+      - name: listCollections
+        object: *database0
+        arguments:
+          filter: { name: { $eq: *collection0Name } }
+        expectResult: [
+          {
+            name: *collection0Name,
+            options: {
+              clusteredIndex: {
+                key: { _id: 1 },
+                unique: true,
+                name: *index0name,
+                v: 2,
+              }
+            }
+          }
+        ]
```

Avoiding JSON-like syntax.

```suggestion
        expectResult:
          - name: *collection0Name
            options:
              clusteredIndex:
                key: { _id: 1 }
                unique: true
                name: *index0Name
                v: 2
```

---

Done.

As a side note, it would be really helpful if there was a style guide for test yaml - there's a remarkably wide variety of styles in use across the existing test files for things like formatting of expectation objects, quoting vs. not, etc. and it's very hard to tell which is the recommended way to do things.

---

https://jira.mongodb.org/browse/DRIVERS-2289